WordPress/Tests/DB/PreparedSQLPlaceholdersUnitTest.inc

```diff
@@ -207,3 +207,24 @@ $wpdb->query( $wpdb->prepare( "DELETE FROM $wpdb->wp_options WHERE option_name L
 $where .= $wpdb->prepare( " AND `name` LIKE '%%%%" . '%s' . "%%%%' ", $args['name'] ); // Bad x 2.
 
 $wpdb->query($wpdb->prepare("delete from wp_postmeta where post_id = $target_postId AND meta_key like 'google_snippets'")); // Bad, the PreparedSQL sniff will also kick in and throw an error about `$target_postId`.
+
+$field = 'post_type';
+$value = 'page';
+
+$wpdb->prepare( 'SELECT ID FROM %i WHERE %i = %s', $wpdb->posts, $field, $value ); // OK.
+$wpdb->prepare( 'SELECT ID FROM %i WHERE %i = %s', $wpdb->posts, $value ); // ReplacementsWrongNumber: "Incorrect number of replacements".
+$wpdb->prepare( 'SELECT ID FROM %i WHERE %i = %x', $wpdb->posts, $field, $value ); // UnsupportedPlaceholder: "Unsupported placeholder used".
+$wpdb->prepare( 'SELECT ID FROM %i WHERE %2$i = %3$s', $wpdb->posts, $field, $value ); // UnquotedComplexPlaceholder: "Complex placeholders used" (not supported with %i, will eventually be disabled with $wpdb->allow_unsafe_unquoted_parameters).
+$wpdb->prepare( 'SELECT ID FROM %i WHERE %i = "%s"', $wpdb->posts, $field, $value ); // QuotedSimplePlaceholder: "Simple placeholders should not be quoted".
+$wpdb->prepare( 'SELECT ID FROM %i WHERE "%i" = %s', $wpdb->posts, $field, $value ); // QuotedSimplePlaceholder: "Simple placeholders should not be quoted".
+$wpdb->prepare( 'SELECT ID FROM %i WHERE `%i` = %s', $wpdb->posts, $field, $value ); // QuotedIdentifierPlaceholder: "Identifier placeholders should not be quoted".
+
```

Comment:
I imagine a test with `%i`.`%i` (i.e. table name, column name format as used with `join`s) would also be a good addition.

Reply:
Yep, done :-)